101-sqlmi-new-vnet/azuredeploy.json

```diff
@@ -0,0 +1,170 @@
+            "metadata": {
+                "description": "Enter managed instance name."
+            }
+        },
+        "administratorLogin": {
+            "type": "string",
+            "defaultValue": "cloudSA",
+            "metadata": {
+                "description": "Enter user name."
+            }
+        },
+        "administratorLoginPassword": {
+            "type": "securestring",
+            "metadata": {
+                "description": "Enter password."
+            }
+        },
+        "location": {
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "Enter location. If you leave this field blank resource group location would be used."
+            }
+        },
+        "virtualNetworkName": {
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "Enter virtual network name. If you leave this field blank name will be created by the template."
+            }
+        },
+        "addressPrefix": {
+            "type": "string",
+            "defaultValue": "10.0.0.0/16",
+            "metadata": {
+                "description": "Enter virtual network address prefix."
+            }
+        },
+        "subnetName": {
+            "type": "string",
+            "defaultValue": "managedInstance",
+            "metadata": {
+                "description": "Enter subnet name."
+            }
+        },
+        "subnetPrefix": {
+            "type": "string",
+            "defaultValue": "10.0.0.0/24",
+            "metadata": {
+                "description": "Enter subnet address prefix."
+            }
+        },
+        "routeTableName": {
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "Enter route table name. If you leave this field blank name will be created by the template."
+            }
+        },
+        "skuName": {
+            "type": "string",
+            "defaultValue": "GP_Gen4",
+            "metadata": {
+                "description": "Enter sku name."
+            }
+        },
+        "storageSizeInGB": {
+            "type": "int",
+            "defaultValue": 32,
+            "metadata": {
+                "description": "Enter storage size."
+            }
+        },
+        "vCores": {
+            "type": "int",
+            "defaultValue": 16,
+            "metadata": {
+                "description": "Enter number of vCores."
+            }
+        },
+        "licenseType": {
+            "type": "string",
+            "defaultValue": "BasePrice",
+            "metadata": {
+                "description": "Enter license type."
+            }
+        }
+    },
+    "variables": {
+        "virtualNetworkName": "[if(equals(parameters('virtualNetworkName'),''), concat('vnet-', parameters('managedInstanceName')), parameters('virtualNetworkName'))]",
```

Comment:
remove all these conditionals and use the defaultValues defined for the param

Reply:
removed vnet and location. route table left as generated but without condition